File: Blind_75/contains-duplicate.java
// 217. Contains Duplicate
// https://leetcode.com/problems/contains-duplicate/

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<Integer>();

        for (int val : nums)
            if (!set.add(val))
                return true;

        return false;
    }
}

File: array/contains_duplicate.js
// 217. Contains Duplicate
// https://leetcode.com/problems/contains-duplicate/

/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
  const set = new Set();

  for (const num of nums) {
    if (set.has(num)) return true;
    set.add(num);
  }

  return false;
};

// since we only want unique values, we use a set
// init a set
// iterate through the nums
// if the num is already in the set, return false
